service/routes.py

```diff
@@ -91,14 +92,37 @@ def create_inventory_records():
     statement += f"and condition: {inventory.condition} created."
     app.logger.info(statement)
     # return jsonify(inventory.serialize()), status.HTTP_201_CREATED
+
     return jsonify(inventory.serialize()), status.HTTP_201_CREATED, {"Location": location_url}
 
 
 @app.route("/inventory", methods=["GET"])
 def list_inventory_records():
     """Returns all of the Inventory records"""
-    app.logger.info("Request list of inventory records")
-    records = Inventory.all()
+    records = []
+    name = request.args.get("name")
+    condition = request.args.get("condition")
+    quantity = request.args.get("quantity")
+    active = request.args.get("active")
+
+    if name:
+        app.logger.info("Filtering by name: %s", name)
+        records = Inventory.find_by_name(name)
+    elif condition:
+        app.logger.info("Filtering by condition:%s", condition)
+        condition = Inventory.Condition(condition)
+        records = Inventory.find_by_condition(condition)
+    elif quantity:
+        app.logger.info("Filtering by quantity: %s", quantity)
+        records = Inventory.find_by_quantity(quantity)
+    elif active:
+        app.logger.info("Filtering by available: %s", active)
+        is_active = active.lower() in ["yes", "y", "true", "t", "1"]
+        records = Inventory.find_by_active(is_active)
+    else:
+        app.logger.info("Request list of all inventory records")
+        records = Inventory.all()
+
```

Comment:
What happens when multiple conditions are passed? For example `active=True AND condition='new'`?

Also do you think we can we push this function to the models file as a helper?

Reply:
as per this function we can only filter on one parameter as professor has done in his lab. As a helper as in?

Reply:
As discussed, let's explore if having multiple filters is within reach, else we may continue with single filter.

So we are kind of deserializing the arguments in this if elif else block right? So if we can reuse that the derialize function somehow or have a similar function to that in the models file so that routes are light on logic. Let me know what you think.